Editor/CoreAPI/GameLiftComputeManager.cs

```diff
@@ -19,7 +22,37 @@ public GameLiftComputeManager(IAmazonGameLiftWrapper wrapper)
             _amazonGameLiftWrapper = wrapper;
         }
 
-        public async Task<RegisterFleetComputeResponse> RegisterFleetCompute(string computeName, string fleetId, string fleetLocation,
+        public async Task<RegisterFleetComputeResponse> RegisterFleetCompute(string computeName, string fleetId,
+            string fleetLocation,
+            string ipAddress)
+        {
+            if (_amazonGameLiftWrapper != null)
```

Comment:
When there are multiple return paths in a method, I find it easiest to follow when using an "if-guard" pattern since it removes nesting. This snippet already uses an "if-guard" pattern for some of the returns, and negating this if's condition can remove the last nesting:
```
if (_amazonGameLiftWrapper == null)
{
    return Response.Fail(new RegisterFleetComputeResponse { ErrorCode = ErrorCode.AccountProfileMissing });
}

if (string.IsNullOrWhiteSpace(computeName))
{
    return Response.Fail(new RegisterFleetComputeResponse { ErrorCode = ErrorCode.InvalidComputeName });
}

if (string.IsNullOrWhiteSpace(ipAddress))
{
    return Response.Fail(new RegisterFleetComputeResponse { ErrorCode = ErrorCode.InvalidIpAddress });
}

return await RegisterCompute(computeName, fleetId, fleetLocation, ipAddress);
```

The other benefit of this pattern is organizational. The most important return value (the nominal path) is last, the most natural place for a return to be, and everything preceding it are edge cases.

Reply:
Yea I agree it does look a lot nicer when browsing through the code. Changed this and Fleet manager to use this pattern.